src/topology/uniform_space/uniform_convergence.lean

```diff
@@ -764,6 +770,84 @@ begin
   exact h.comp _ (maps_to_univ _ _) cg
 end
 
+lemma tendsto_locally_uniformly_on_tfae [locally_compact_space α]
+  (G : ι → α → β) (g : α → β) (p : filter ι) (hs : is_open s) :
+  [ (tendsto_locally_uniformly_on G g p s),
+    (∀ K ⊆ s, is_compact K → tendsto_uniformly_on G g p K),
+    (∀ x ∈ s, ∃ v ∈ 𝓝[s] x, tendsto_uniformly_on G g p v) ].tfae :=
+begin
+  tfae_have : 1 → 2,
+  { rintro h K hK1 hK2,
+    exact (tendsto_locally_uniformly_on_iff_tendsto_uniformly_on_of_compact hK2).mp (h.mono hK1) },
+  tfae_have : 2 → 3,
+  { rintro h x hx,
+    obtain ⟨K, ⟨hK1, hK2⟩, hK3⟩ := (compact_basis_nhds x).mem_iff.mp (hs.mem_nhds hx),
+    refine ⟨K, nhds_within_le_nhds hK1, h K hK3 hK2⟩ },
+  tfae_have : 3 → 1,
+  { rintro h u hu x hx,
+    obtain ⟨v, hv1, hv2⟩ := h x hx,
+    exact ⟨v, hv1, hv2 u hu⟩ },
+  tfae_finish
+end
+
+lemma tendsto_locally_uniformly_on_iff_forall_compact [locally_compact_space α] (hs : is_open s) :
+  tendsto_locally_uniformly_on F f p s ↔
+  ∀ K ⊆ s, is_compact K → tendsto_uniformly_on F f p K :=
+(tendsto_locally_uniformly_on_tfae F f p hs).out 0 1
+
+lemma tendsto_locally_uniformly_on_iff_filter :
+  tendsto_locally_uniformly_on F f p s ↔
+  ∀ x ∈ s, tendsto_uniformly_on_filter F f p (𝓝[s] x) :=
+begin
+  simp only [tendsto_uniformly_on_filter, eventually_prod_iff],
+  split,
+  { rintro h x hx u hu,
+    obtain ⟨s, hs1, hs2⟩ := h u hu x hx,
+    exact ⟨_, hs2, _, eventually_of_mem hs1 (λ x, id), λ i hi y hy, hi y hy⟩ },
+  { rintro h u hu x hx,
+    obtain ⟨pa, hpa, pb, hpb, h⟩ := h x hx u hu,
+    refine ⟨pb, hpb, eventually_of_mem hpa (λ i hi y hy, h hi hy)⟩ }
+end
+
+lemma tendsto_locally_uniformly_iff_filter :
+  tendsto_locally_uniformly F f p ↔
+  ∀ x, tendsto_uniformly_on_filter F f p (𝓝 x) :=
+by simpa [← tendsto_locally_uniformly_on_univ, ← nhds_within_univ] using
+    @tendsto_locally_uniformly_on_iff_filter _ _ _ _ F f univ p _
+
+lemma tendsto_locally_uniformly_on.tendsto_at (hf : tendsto_locally_uniformly_on F f p s)
+  ⦃a : α⦄ (ha : a ∈ s) :
```

Comment:
The `⦃a : α⦄` version enables to say
```
have : \forall a \in s, whatever := tendsto_locally_uniformly_on.tendsto_at hf
```
rather than
```
have : \forall a \in s, whatever := \lambda a ha, tendsto_locally_uniformly_on.tendsto_at hf ha
```
which looks slightly nicer. I won't change the existing version for fear of breaking things, but I can amend mine.